erdjs/src/smartcontracts/interaction/interactor.spec.ts

```diff
@@ -0,0 +1,199 @@
+import { SmartContractInteractor } from "./interactor";
+import { StrictChecker } from "./strictChecker";
+import { DefaultInteractionRunner } from "./defaultRunner";
+import { SmartContract } from "../smartContract";
+import { AbiRegistry, missingOption, providedOption, typedBigInt, typedUTF8, U32Value } from "../typesystem";
+import { AddImmediateResult, loadAbiRegistry, MarkNotarized, MockProvider, setupUnitTestWatcherTimeouts, TestWallets, Wait } from "../../testutils";
+import { SmartContractAbi } from "../abi";
+import { Address } from "../../address";
+import { assert } from "chai";
+import { Interaction } from "./interaction";
+import { GasLimit } from "../../networkParams";
+import { ContractFunction } from "../function";
+import { QueryResponse } from "../queryResponse";
+import { Nonce } from "../../nonce";
+import { TransactionStatus } from "../../transaction";
+import { ReturnCode } from "../returnCode";
+import { Balance } from "../../balance";
+import BigNumber from "bignumber.js";
+
+describe("test smart contract interactor", function () {
+    let wallets = new TestWallets();
+    let dummyAddress = new Address("erd1qqqqqqqqqqqqqpgqak8zt22wl2ph4tswtyc39namqx6ysa2sd8ss4xmlj3");
+    let checker = new StrictChecker();
+    let provider = new MockProvider();
+    let signer = wallets.alice.signer;
+    let runner = new DefaultInteractionRunner(checker, signer, provider);
+
+    it("should interact with 'answer'", async function () {
+        setupUnitTestWatcherTimeouts();
+
+        let abiRegistry = await loadAbiRegistry(["src/testdata/answer.abi.json"]);
```

Comment:
I think the setup is a little verbose for developers, we need to instantiate 4 components in order to be able to call sc functions (5, counting the runner).

Reply:
That's right. I'll make a (configurable) factory, with decent defaults.

Reply:
Simplified (no factory, but removed the `interactor` object). Is it all right?